src/consumer_queue.cpp

```diff
@@ -35,12 +37,10 @@ sample_p consumer_queue::pop_sample(double timeout) {
 		buffer_.pop(result);
 	} else {
 		if (!buffer_.pop(result)) {
-			// turn timeout into the point in time at which we give up
-			timeout += lsl::lsl_clock();
-			do {
-				if (lsl::lsl_clock() >= timeout) break;
-				std::this_thread::sleep_for(std::chrono::milliseconds(1));
-			} while (!buffer_.pop(result));
+			// wait untill for a new sample until the thread calling push_sample delivers one, or until timeout
+			std::unique_lock<std::mutex> lk(lock_);
```

Comment:
If we move this lock to the top of the function, we can have it be thread-safe in the presence of a buffer overflow.